src/address.rs

```diff
@@ -1,91 +1,338 @@
+              A: Handler<M> + Send,
+              S: Stream<Item = M> + Send + Unpin + 'static,
+              Self: Sized + Send + 'static,
+    {
+        #[cfg(feature = "with-tokio-0_2")]
+        tokio::spawn(async move {
+            while let Some(msg) = stream.next().await {
+                if let Err(_) = self.do_send(msg) {
+                    break;
+                }
+            }
+        });
+
+        #[cfg(feature = "with-async_std-1")]
+        async_std::task::spawn(async move {
+            while let Some(msg) = stream.next().await {
+                if let Err(_) = self.do_send(msg) {
+                    break;
+                }
+            }
+        });
+    }
+
+    /// Attaches a stream to this actor such that all messages produced by it are forwarded to the
+    /// actor and handled asynchronously. This could, for instance, be used to forward messages from a
+    /// socket to the actor (after the messages have been appropriately `map`ped). This is a
+    /// convenience method over a manual async stream forwarding loop.
+    #[doc(cfg(feature = "with-tokio-0_2"))]
+    #[doc(cfg(feature = "with-async_std-1"))]
+    #[cfg(any(doc, feature = "with-tokio-0_2", feature = "with-async_std-1"))]
+    fn attach_stream_async<S, M>(self, mut stream: S)
+        where M: Message,
+              A: AsyncHandler<M> + Send,
+              S: Stream<Item = M> + Send + Unpin + 'static,
+              Self: Sized + Send + 'static,
+    {
+        #[cfg(feature = "with-tokio-0_2")]
+        tokio::spawn(async move {
+            while let Some(msg) = stream.next().await {
+                if let Err(_) = self.do_send_async(msg) {
+                    break;
+                }
+            }
+        });
+
+        #[cfg(feature = "with-async_std-1")]
+        async_std::task::spawn(async move {
+            while let Some(msg) = stream.next().await {
+                if let Err(_) = self.do_send_async(msg) {
+                    break;
+                }
+            }
+        });
+    }
+}
+
+/// An `Address` is a reference to an actor through which [`Message`s](trait.Message.html) can be
+/// sent. It can be cloned, and when all `Address`es are dropped, the actor will be stopped. Therefore,
+/// any existing `Address`es will inhibit the dropping of an actor. If this is undesirable, then
+/// the [`WeakAddress`](struct.WeakAddress.html) struct should be used instead. This struct is created
+/// by calling the [`Actor::start`](trait.Actor.html#method.start) or  [`Actor::spawn`](trait.Actor.html#method.start)
+/// methods.
+pub struct Address<A: Actor> {
+    pub(crate) sender: UnboundedSender<ManagerMessage<A>>,
+    pub(crate) ref_counter: Arc<()>,
+}
+
+impl<A: Actor> Address<A> {
+    /// Create a weak address to the actor. Unlike with the strong variety of address (this kind),
+    /// an actor will not be prevented from being dropped if only weak addresses exist.
+    pub fn weak(&self) -> WeakAddress<A> {
```

Comment:
What about using same naming scheme as `Rc` for weak addresses? That being `upgrade` and `downgrade`?

Reply:
How would the two (into_weak and just weak) be accommodated?

Reply:
`downgrade` and `into_downgraded`